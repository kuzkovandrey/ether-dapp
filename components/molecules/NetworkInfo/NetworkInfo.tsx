import { Badge, Card, Flex, Heading } from '@radix-ui/themes';
import { ReactNode } from 'react';

export type NetworkInfoProps = {
  name: string;
  isConnected: boolean;
  actionSlot?: ReactNode;
};

function NetworkInfo({ name, isConnected, actionSlot }: NetworkInfoProps) {
  return (
    <Card data-testid="network-info" style={{ width: '100%' }}>
      <Flex direction="column" gap="4">
        <Flex justify="between" align="center">
          <Heading data-testid="network-info-heading" size="3">
            Network: {name}
          </Heading>
          <Flex gap="2" align="center">
            <Badge data-testid="network-info-badge" size="2" color={isConnected ? 'green' : 'red'}>
              {isConnected ? 'Connected' : 'Not connected'}
            </Badge>
          </Flex>
        </Flex>
        {actionSlot && <div data-testid="network-info-action-slot">{actionSlot}</div>}
      </Flex>
    </Card>
  );
}

export default NetworkInfo;
