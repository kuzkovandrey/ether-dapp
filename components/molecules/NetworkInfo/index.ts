export { default as NetworkInfo } from './NetworkInfo';
