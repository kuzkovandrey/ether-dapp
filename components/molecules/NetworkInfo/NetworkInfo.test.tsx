/**
 * @jest-environment jsdom
 */

import { cleanup, render, screen } from '@testing-library/react';

import '@testing-library/jest-dom';
import NetworkInfo, { NetworkInfoProps } from './NetworkInfo';

afterEach(cleanup);

describe('NetworkInfo component', () => {
  let props = {} as NetworkInfoProps;

  beforeEach(() => {
    props = {
      name: 'Eth mainnet',
      isConnected: false,
    };
  });

  test('# component should be rendered on the screen', () => {
    render(<NetworkInfo {...props} />);
    expect(screen.getByTestId('network-info')).toBeInTheDocument();
    expect(screen.getByTestId('network-info-heading')).toHaveTextContent(`Network: ${props.name}`);
  });

  test('# component must show connection status', () => {
    const { rerender } = render(<NetworkInfo {...props} />);

    expect(screen.getByTestId('network-info-badge')).toHaveTextContent(`Not connected`);

    props.isConnected = true;
    rerender(<NetworkInfo {...props} />);
    expect(screen.getByTestId('network-info-badge')).toHaveTextContent(`Connected`);
  });

  test('# component must have action contect', () => {
    const { rerender } = render(<NetworkInfo {...props} />);

    expect(screen.queryByTestId('network-info-action-slot')).not.toBeInTheDocument();

    props.actionSlot = <p>Action</p>;
    rerender(<NetworkInfo {...props} />);
    expect(screen.queryByTestId('network-info-action-slot')).toBeInTheDocument();
    expect(screen.queryByTestId('network-info-action-slot')).toMatchSnapshot();
  });
});
