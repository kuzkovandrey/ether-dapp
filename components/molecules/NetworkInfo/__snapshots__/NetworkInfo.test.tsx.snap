// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NetworkInfo component # component must have action contect 1`] = `
<div
  data-testid="network-info-action-slot"
>
  <p>
    Action
  </p>
</div>
`;
